buildSrc/src/main/kotlin/tasks/ExtractAndroidStudioTask.kt

```diff
@@ -0,0 +1,85 @@
+package tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileSystemOperations
+import org.gradle.api.file.RelativePath
+import org.gradle.api.internal.file.FileOperations
+import org.gradle.api.tasks.*
+import org.gradle.process.ExecOperations
+import org.gradle.work.DisableCachingByDefault
+import java.io.File
+import javax.inject.Inject
+
+
+@DisableCachingByDefault(because = "Not worth caching")
+abstract class ExtractAndroidStudioTask @Inject constructor(
```

Comment:
💭 This is the same implementation as we have in gradle/gradle https://github.com/gradle/gradle/blob/81c9b411450ccb9a5c52df2ef77459dad08fddb4/build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/ide/ExtractAndroidStudioTask.kt#L38